### PR TITLE
Decentralizing: Merging Getter and Setter

### PR DESCRIPTION
**Pull request by Yoric**

### CODE REVIEW

File: components/openzwave-adapter/src/lib.rs [hunk 1]
```diff
@@ -302,36 +302,28 @@ impl OpenzwaveAdapter {
                         let kind = kind.unwrap();
 
                         if has_getter {
-                            let getter_id = TaxoId::new(&value_id);
+                            let getter_id = TaxoId::<Channel>::new(&value_id);
```
Comment (julienw) [inline components/openzwave-adapter/src/lib.rs:305]: are you sure this is needed ? Isn't it infered by the subsequent push ?

File: components/openzwave-adapter/src/lib.rs [hunk 2]
```diff
@@ -302,36 +302,28 @@ impl OpenzwaveAdapter {
...
-                                }
+                                kind: kind.clone(),
                             }).unwrap_or_else(|e| {
                                 error!("Couldn't add the getter {}: {}", value_id, e);
                             });
                         }
 
                         if has_setter {
-                            let setter_id = TaxoId::new(&value_id);
+                            let setter_id = TaxoId::<Channel>::new(&value_id);
```
Comment (julienw) [inline components/openzwave-adapter/src/lib.rs:319]: same question

File: components/openzwave-adapter/src/lib.rs [hunk 3]
```diff
@@ -182,8 +182,8 @@ pub struct OpenzwaveAdapter {
     version: [u32; 4],
     ozw: ZWaveManager,
     node_map: IdMap<ServiceId, Node>,
-    getter_map: IdMap<Getter, ValueID>,
-    setter_map: IdMap<Setter, ValueID>,
+    getter_map: IdMap<Channel, ValueID>,
+    setter_map: IdMap<Channel, ValueID>,
```
Comment (julienw) [inline components/openzwave-adapter/src/lib.rs:186]: I think we'll want to merge these maps into the same map; probably more than you're willing to do in this patch but please file a bug.

**Comment by julienw** [comment]
A few comments on the zwave part; you'll need a review for the other parts though.

File: components/openzwave-adapter/src/lib.rs (hunk 3 above)
Comment (Yoric) [inline components/openzwave-adapter/src/lib.rs:186]: I'm currently writing the patch to do this in Taxonomy, so yeah, that's planned.

File: components/openzwave-adapter/src/lib.rs (hunk 1 above)
Comment (Yoric) [inline components/openzwave-adapter/src/lib.rs:305]: I'll double-check.

File: components/openzwave-adapter/src/lib.rs (hunk 3 above)
Comment (julienw) [inline components/openzwave-adapter/src/lib.rs:186]: But here we're not in taxonomy ? Will you do this here as well ?

Comment (Yoric) [inline components/openzwave-adapter/src/lib.rs:186]: Not 100% sure. I'll file an issue, as you suggested.

**Comment by dhylands** [comment]
r+ for ip_camera related changes

**Comment by Yoric** [comment]
Also, f? @mcav if you wish to see in which direction Decentralized Taxonomy is heading.

**Comment by fabricedesre** [comment]
r=me for the adapter/tts/ changes

**Pull request merged**